src/plugins/event-plugins/DomEventPlugin.ts

```diff
@@ -129,13 +129,30 @@ export class DomEventPlugin implements Plugin {
             };
             if (cssLocator !== undefined) {
                 eventData.cssLocator = cssLocator;
+                const elementRumId = this.getElementRumId(event, cssLocator);
+                if (elementRumId) {
+                    eventData.elementRumId = elementRumId;
+                }
             }
             if (this.recordEvent) {
                 this.recordEvent(DOM_EVENT_TYPE, eventData);
             }
         };
     }
 
+    private getElementRumId(event: Event, cssLocator: string): string {
+        if (event.composedPath) {
+            const eventPath = event.composedPath();
+            const firstMatchingElement = eventPath.find((element: Element) =>
+                element.matches(cssLocator)
+            ) as Element;
+            if (firstMatchingElement.hasAttribute('data-rum-id')) {
+                return firstMatchingElement.getAttribute('data-rum-id');
+            }
+        }
+        return '';
```

Comment:
1. Can we return `null` so that we only add the `elementRumId` data to the `eventData` if an element with the `data-rum-id` attribute was found?
2. Could I know more about your use case? This implementation would require us to update the dom event type schema to add an additional field.
3. What are the performance implications if we were to look for all elements with the `data-rum-id` attribute for each element identified by the provided `cssLocator` ? Would this require looking through the entire DOM until one/the closest one is found?

Reply:
Hello Grace, thank you for your time reviewing this.
1. There is a validation for this on line 133, this would return falsy for empty string and would not add elementRumId.
`if (elementRumId) {`

2. I would like to capture user clicks accurately on my website which uses React on the front end. Many of the UI libraries creates child elements while rendering the page. And as a developer I don't have access to these child elements. I am unable to provide an id or a css selector. When a users clicks on a button, I either get "SPAN" or a random id that was generated during the render instead of the id / cssLocator that I provided to the Button component. This is a common issue with the current structure. This PR would solve this problem.

3. This is a great call out. I would not expect a significant impact on the performance. event.composedPath() returns an array that lists parents of the clicked element vertically. However we can add a limit for this, let's say 5 levels above or something like that. Let me know what you think, please.